internal/provider/cluster_resource_test.go

```diff
@@ -685,28 +692,55 @@ func TestIntegrationDedicatedClusterResource(t *testing.T) {
 	s.EXPECT().GetCluster(gomock.Any(), clusterID).
 		Return(&finalizedCluster, httpOk, nil).Times(3)
 
+	// First Update
 	// Scale
 
 	s.EXPECT().UpdateCluster(gomock.Any(), clusterID, gomock.Any()).
 		DoAndReturn(func(context.Context, string, *client.UpdateClusterSpecification,
 		) (*client.Cluster, *http.Response, error) {
-			currentCluster := &scaledCluster
+			currentCluster := &firstUpdateCluster
 			return currentCluster, httpOk, nil
 		})
 
 	s.EXPECT().GetCluster(gomock.Any(), clusterID).
-		Return(&scaledCluster, httpOk, nil).Times(5)
+		Return(&firstUpdateCluster, httpOk, nil).Times(5)
+
+	// Failed Delete Attempt
+	s.EXPECT().GetCluster(gomock.Any(), clusterID).
+		Return(&firstUpdateCluster, httpOk, nil).Times(2)
+
+	// Second Update
+	s.EXPECT().UpdateCluster(gomock.Any(), clusterID, gomock.Any()).
+		DoAndReturn(func(context.Context, string, *client.UpdateClusterSpecification,
+		) (*client.Cluster, *http.Response, error) {
+			currentCluster := &secondUpdateCluster
+			return currentCluster, httpOk, nil
+		})
+
+	s.EXPECT().GetCluster(gomock.Any(), clusterID).
+		Return(&secondUpdateCluster, httpOk, nil).Times(5)
 
 	// Deletion
 
 	s.EXPECT().DeleteCluster(gomock.Any(), clusterID)
 
-	scaleStep := resource.TestStep{
-		Config: getTestDedicatedClusterResourceConfig(clusterName, latestClusterMajorVersion, false, 4),
+	firstUpdateStep := resource.TestStep{
+		Config: getTestDedicatedClusterResourceConfig(clusterName, latestClusterMajorVersion, false, 4, ptr(true)),
 		Check:  resource.TestCheckResourceAttr("cockroach_cluster.dedicated", "dedicated.num_virtual_cpus", "4"),
 	}
 
-	testDedicatedClusterResource(t, clusterName, true, scaleStep)
+	failedDeleteStep := resource.TestStep{
+		Config:      " ",
+		Destroy:     true,
+		ExpectError: regexp.MustCompile(".*Cannot destroy cluster with delete protection enabled*"),
+	}
+
```

Comment:
It's kind of unfortunate this step is not getting tested via the acceptance tests.  can it be added somewhere more general?

Reply:
Oh, ya I think I can make that work. Let me give it a try.